### PR TITLE
fix phone number parser (DEV-1949)

### PR DESCRIPTION
**Pull request by mikefeldberg**
DEV-1949

## Summary by Sourcery

Treat blank phone number inputs as null in the GraphQL scalar and extend mutation tests to cover and persist empty phone numbers.

Bug Fixes:
- ParseValue lambda now returns None for empty or whitespace-only phone number inputs instead of attempting to parse them.

Tests:
- Add a test case for an empty string phone number in the update mutation validation.
- Assert that the database stored phone number matches the expected value after mutation.

### CODE REVIEW

File: apps/betterangels-backend/common/graphql/types.py [hunk 1]
```diff
@@ -54,7 +54,7 @@ def _serialize_phone_number(v: DjangoPhoneNumber) -> str:
 
 PhoneNumberScalar: DjangoPhoneNumber | str = strawberry.scalar(
     DjangoPhoneNumber,
-    parse_value=lambda v: _parse_phone_number(v.strip()),
+    parse_value=lambda v: _parse_phone_number(v.strip()) if v.strip() else None,
```
[the remark this answers is not in the document]
Comment (mikefeldberg) [reply apps/betterangels-backend/common/graphql/types.py:56]: updated the serializer. The updated parser should make storing an empty string functionally impossible (django admin saves as null), but this is safer. No need for clean up. I checked and we have no contacts with string phone numbers. And, even if we did, this serializer would handle them
